experiments/yarn/src/main/scala/io/gearpump/experiments/yarn/client/Client.scala

```diff
@@ -17,323 +17,171 @@
-    system.scheduler.scheduleOnce(Duration(5, TimeUnit.SECONDS)) {
-      LOG.info(s"host=${trackingURL.getHost} port=${trackingURL.getPort} uri=/proxy/${report.getApplicationId}/$restURI/master")
-      lazy val trackingURLConnectionFlow: Flow[HttpRequest, HttpResponse, Any] =
-        Http().outgoingConnection(trackingURL.getHost, trackingURL.getPort)
-      val response = Source.single(RequestBuilding.Get(s"/proxy/${report.getApplicationId}/$restURI/master")).via(trackingURLConnectionFlow).runWith(Sink.head)
-      response.foreach(response => {
-        response.status match {
-          case StatusCodes.OK =>
-            LOG.info(s"status code=${StatusCodes.OK.intValue}")
-            Unmarshal(response.entity).to[String].onComplete(result => {
-              result match {
-                case Success(data) =>
-                  val masterData = read[MasterData](data)
-                  LOG.info(s"leader=${masterData.masterDescription.leader._1}:${masterData.masterDescription.leader._2}")
-                  val cluster=masterData.masterDescription.cluster.map(p=>{p._1+":"+p._2}).mkString(",")
-                  LOG.info("masters=" + cluster)
-                case Failure(throwable) =>
-                  LOG.error("Failed to fetch masters", throwable)
-              }
-              system.shutdown()
-            })
-          case value =>
-            LOG.error(s"Bad status code=${value.intValue}")
-            system.shutdown()
-        }
-      })
-    }
-  }
+  private def uploadConfigToHDFS(appId: ApplicationId): String = {
+    // will use personal home directory so that it will not conflict with other users
+    val confDir = new Path(fs.getHomeDirectory(), s".gearpump_app${appId.getId}/conf/")
 
-  def start(): Boolean = {
-    Try({
-      yarnClient.init(yarnConf)
-      yarnClient.start()
-      true
-    }) match {
-      case Success(success) =>
-        success
-      case Failure(throwable) =>
-        LOG.error("Failed to start", throwable)
-        false
-    }
-  }
+    // copy config from local to remote.
+    val remoteConfFile = new Path(confDir, "gear.conf")
+    var out = fs.create(remoteConfFile)
+    var writer = new OutputStreamWriter(out)
 
-  def submit(): Try[ApplicationId] = {
-    Try({
-      val appContext = yarnClient.createApplication.getApplicationSubmissionContext
-      appContext.setApplicationName(getEnv(YARNAPPMASTER_NAME))
+    val filterJvmReservedKeys = ClusterConfig.filterOutJvmReservedKeys(akka)
 
-      val containerContext = containerLaunchContext(getCommand)
-      appContext.setAMContainerSpec(containerContext)
-      appContext.setResource(getAMCapability)
-      appContext.setQueue(getEnv(YARNAPPMASTER_QUEUE))
+    writer.write(filterJvmReservedKeys.root().render())
+    writer.close()
+    out.close()
 
-      yarnClient.submitApplication(appContext)
-      appContext.getApplicationId
-    })
-  }
+    // save yarn-site.xml to remote
```

Comment:
@kkasravi, GearPump broker runs yarnclient on Cloud Foundry. I assumed yarn-site.xml to be present on machine/container where yarnclient is. I planned to supply yarn-site.xml with the broker app.